src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1885,6 +1926,140 @@ class module_sram : public module_impl
+    auto get_dtrace_buffer_size =
+        (get_dtrace_buffer_size_fun) xrt_core::dlsym(dtrace.lib_hdl->handle, "get_dtrace_buffer_size");
+    if (!get_dtrace_buffer_size) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module", xrt_core::dlerror());
+      return;
+    }
+
+    using create_dtrace_buffer_fun = void (*)(const char*, const char*, size_t, uint64_t*, size_t, uint64_t*);
+    auto create_dtrace_buffer =
+        (create_dtrace_buffer_fun) xrt_core::dlsym(dtrace.lib_hdl->handle, "create_dtrace_buffer");
+    if (!create_dtrace_buffer) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module", xrt_core::dlerror());
+      return;
+    }
+
+    uint32_t ctrl_buf_size;
+    uint32_t mem_buf_size;
+    if (get_dtrace_buffer_size(dtrace.ctrl_file_path.c_str(), dtrace.map_data.c_str(),
+                               &ctrl_buf_size, &mem_buf_size) != 0) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
+                              "[dtrace] : Failed to get control buffer size");
+      return;
+    }
+
+    if (!ctrl_buf_size) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
+                              "[dtrace] : Control buffer size is zero, no dtrace o/p");
+      return;
+    }
+
+    std::string msg;
+    try {
+      // below call creates dtrace xrt control buffer and informs driver / firmware with the buffer address
+      m_dtrace_ctrl_bo = xrt_core::bo_int::create_dtrace_bo(m_hwctx, ctrl_buf_size * sizeof(uint32_t));
+      // also create dtrace mem buffer if size is non zero
+      if (mem_buf_size) {
+        m_dtrace_mem_bo = xrt::bo{ m_hwctx, mem_buf_size * sizeof(uint32_t), xrt::bo::flags::cacheable, 1 /* fix me */ };
+        // fill data by calling dtrace library API with bo map ptr and sync the bo
+        create_dtrace_buffer(dtrace.ctrl_file_path.c_str(), dtrace.map_data.c_str(),
+                             ctrl_buf_size, m_dtrace_ctrl_bo.map<uint64_t*>(),
+                             mem_buf_size, m_dtrace_mem_bo.map<uint64_t*>());
+
+        m_dtrace_mem_bo.sync(XCL_BO_SYNC_BO_TO_DEVICE);
+      }
+      else {
+        // fill data by calling dtrace library API without mem buffer
+        create_dtrace_buffer(dtrace.ctrl_file_path.c_str(), dtrace.map_data.c_str(),
+                             ctrl_buf_size, m_dtrace_ctrl_bo.map<uint64_t*>(),
+                             0, nullptr);
+      }
+
+      // sync dtrace control buffer
+      m_dtrace_ctrl_bo.sync(XCL_BO_SYNC_BO_TO_DEVICE);
+      msg = "[dtrace] : dtrace buffers initialized successfully";
+    }
+    catch (const std::exception &e) {
+      m_dtrace_ctrl_bo = {};
+      msg = std::string{"[dtrace] : dtrace buffers initialization failed, "} + e.what();
+    }
+
+    xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module", msg);
```

Comment:
Remove and put on line 2053 and 2057.  Move std::string from line 2030 to local variable if at all needed.